packages/expo-image-picker/ios/MediaHandler.swift

```diff
@@ -259,4 +346,42 @@ internal struct MediaHandler {
       return nil
     }
   }
+
+  private func getFileExtension(from fileName: String) -> String {
+    return ".\(URL(fileURLWithPath: fileName).pathExtension)"
+  }
+}
+
+fileprivate extension NSItemProvider {
```

Comment:
Let's put this in a separate file called `NSItemProvider+Async`. We also have an async version for `loadDataRepresentation` that can be moved to the extension too.